python/ray/data/_internal/progress_bar.py

```diff
@@ -158,9 +158,15 @@ def fetch_until_complete(self, refs: List[ObjectRef]) -> List[Any]:
             )
             if fetch_local:
                 fetch_local = False
+            total_rows_processed = 0
             for ref, result in zip(done, ray.get(done)):
                 ref_to_result[ref] = result
-            self.update(len(done))
+                num_rows = (
+                    result.num_rows if hasattr(result, "num_rows") else 1
+                )  # Default to 1 if no row count is available
+                total_rows_processed += num_rows
+            # TODO(zhilong): Change the total to total_row when init progress bar
+            self.update(total_rows_processed)
```

Comment:
nice, thanks for the fix here. for consistency, can we also apply the same logic in `block_until_complete()` here? https://github.com/ray-project/ray/pull/46924/files/affadbab90fdb47b8838f0dc0f40d917ccbd45c3..e231e41c1d436434b7c20b2b8203bc254e3828d2#diff-968882a8882d3516ef0f814415b69f55edd379743ee4b8a9a1c849fc1afede04R137

Reply:
fixed